apps/web/src/components/AssessmentStepTracker/utils.ts

```diff
@@ -184,11 +190,257 @@ export const sortResultsAndAddOrdinal = (
   return resultsWithOrdinal.sort((resultA, resultB) => {
     return (
       getBookmarkValue(resultB) - getBookmarkValue(resultA) ||
-      getDecisionValue(resultA) - getDecisionValue(resultB) ||
+      getDecisionValue(resultA.decision) - getDecisionValue(resultB.decision) ||
       getPriorityValue(resultB) - getPriorityValue(resultA) ||
       getVeteranValue(resultB) - getVeteranValue(resultA) ||
       compareLastNames(resultA, resultB) ||
       compareFirstNames(resultA, resultB)
     );
   });
 };
+
+const getResultsDecision = (
+  step: AssessmentStep,
+  results?: AssessmentResult[],
+): NullableDecision => {
+  if (!results) return NO_DECISION;
+  let hasFailure: boolean = false;
+  let hasOnHold: boolean = false;
+  let hasToAssess: boolean = false;
+
+  const stepResults = results.filter((result) => {
+    return result.assessmentStep?.id === step.id;
+  });
+
+  if (stepResults.length === 0) {
+    hasToAssess = true;
+  }
+
+  const requiredSkillAssessments = step.poolSkills?.filter(
+    (poolSkill) => poolSkill?.type === PoolSkillType.Essential,
+  );
+
+  requiredSkillAssessments?.forEach((skillAssessment) => {
+    const assessmentResults = stepResults.filter((result) => {
+      return result.poolSkill?.id === skillAssessment?.id;
+    });
+
+    if (assessmentResults.length === 0) {
+      hasToAssess = true;
+      return;
+    }
+
+    assessmentResults.forEach((assessmentResult) => {
+      switch (assessmentResult.assessmentDecision) {
+        case null:
+          hasToAssess = true;
+          break;
+        case AssessmentDecision.Hold:
+          hasOnHold = true;
+          break;
+        case AssessmentDecision.Unsuccessful:
+          hasFailure = true;
+          break;
+        default:
+      }
+    });
+  });
+
+  // Could be an education requirement that is not an essential skill
+  stepResults.forEach((result) => {
+    // Any "to assess" should be marked
+    if (result.assessmentDecision === null) {
+      hasToAssess = true;
+    }
+
+    if (result.assessmentResultType === AssessmentResultType.Education) {
+      switch (result.assessmentDecision) {
+        case null:
+          hasToAssess = true;
+          break;
+        case AssessmentDecision.Hold:
+          hasOnHold = true;
+          break;
+        case AssessmentDecision.Unsuccessful:
+          hasFailure = true;
+          break;
+        default:
+      }
+    }
+  });
```

Comment:
This only applies to the APPLICATION_SCREENING assessment type

Reply:
Oh I was basing this off of [the logic outlined](https://github.com/GCTC-NTGC/gc-digital-talent/issues/8957#:~:text=Essential%20requirements%20%3D%20essential%20skills%20AND%20education%20requirements) 😅 

So, should it be `step.type === AssessmentStepType.ApplicationScreening` instead?

Reply:
Fixed in [fix step decision logic](https://github.com/GCTC-NTGC/gc-digital-talent/pull/9022/commits/1b0d12688ce81ae146853389288f1682058bc124)